### DISCUSSION TITLE
[Feature Request] Allow dynamic insertion (templating) of basic variables (user/org/repo/branch name) into markdown

### DISCUSSION THREAD

**Discussion by vincerubinetti** [community/community#37122] [Other Feature Feedback, Questions, & Ideas]
Please upvote 👍 so this has a better chance of getting some attention.

Related Stack Overflow: https://stackoverflow.com/questions/40196198/automatic-link-to-github-pages-url-from-readme-md

When you are making a repo that is meant to be forked or templated, you often want to write the readme instructions in a way that is generic so that when a user clones it, they see instructions/links/etc that match their own org and repo. One major example would be linking to a GitHub Pages site in the readme, like `Once you enable GitHub Pages, your site should be live at <user>.github.io/<repo>`. But there are many other cases where this would be useful, like linking to the `.dev` editor, or more cleanly linking to another assumed repo in the same org.

The variables I can imagine would be most useful are `org_name`, `repo_name`, `current_branch`, `pages_url` (could be `github.io` or custom domain).

There is a GitHub Action here that fills this need, but it has to be run any time there is a change, and also needs to make a new commit: https://github.com/marketplace/actions/dynamic-readme 

I'm sure this has been requested before somewhere, but I could not find it.

**Comment by itcarroll** [comment]
This is my most upvoted question on SO!

Could it be done with an enhancement to autolinking? Since `Username/Repository#Issue` is already a valid shortlink in conversations, and [custom autolinking](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls#custom-autolinks-to-external-resources) exists, it's hard to believe `.md` files could not be outfitted with these "variables".

**Comment by ossanna16** [comment]
Hi there @vincerubinetti and welcome to our community! Thank you for your suggestion 🙂

**Comment by abernier** [comment]
It is even more useful for [template](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository)/forked repos, where for instance, badges links need to be org/user relative, eg for `jdoe/superfun`:

```md
[Edit me on Codesandbox](https://codesandbox.io/s/github/jdoe/superfun)
```

would make a lot of sense being able to dynamically print `jdoe` and `superfun` in the md link

**Reply by abernier** [reply]
or even for [Github workflow status badges](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge#using-the-workflow-file-name)

```md
https://github.com/<OWNER>/<REPOSITORY>/actions/workflows/<WORKFLOW_FILE>/badge.svg
```
